### PR TITLE
Function '_reverse_from_offset' returns pair of line and its offset

### PR DESCRIPTION
**Pull request by konefalg**

### CODE REVIEW

File: whylog/client/searchers.py [hunk 1]
```diff
@@ -40,6 +44,7 @@ def _reverse_from_offset(self, offset, buf_size=BufsizeConsts.STANDARD_BUF_SIZE)
             fh.seek(offset)
```
Comment (andrzejgorski) [inline whylog/client/searchers.py]: Don't use shortcuts like fh. Name file_path is much readable.

File: whylog/client/searchers.py [hunk 2]
```diff
@@ -31,6 +31,10 @@ def __init__(self, file_path):
     def search(self, search_data):
         pass
 
+    @classmethod
+    def _decrease_actual_offset_properly(cls, actual_offset, decreasion_string):
+        return actual_offset - len(decreasion_string) - 1
+
     def _reverse_from_offset(self, offset, buf_size=BufsizeConsts.STANDARD_BUF_SIZE):
```
Comment (andrzejgorski) [inline whylog/client/searchers.py]: This function is too long split apart it, if it is possible. We need to meet and talk about this function.

File: whylog/tests/tests_client/test_backtrackerSearcher.py [hunk 3]
```diff
@@ -36,12 +36,20 @@ def _verify_lines(self, lines_normally, lines_reversed):
             assert i == j
 
     def _run_reverse_and_check_results(
-        self, log_file_path, reverse_from_offset_params, how_many_last_lines
+        self,
+        log_file_path,
+        reverse_from_offset_params,
+        how_many_last_lines,
+        final_offset_should_be_zero=True
```
Comment (andrzejgorski) [inline whylog/tests/tests_client/test_backtrackerSearcher.py]: Use double identation for function arguments.

```
def _run_reverse_and_check_results(
        self,
        log_file_path,
        reverse_from_offset_params,
        how_many_last_lines,
        final_offset_should_be_zero=True 
):
    backtracker = searchers.BacktrackSearcher(log_file_path)
```

File: whylog/client/searchers.py (hunk 2 above)
Comment (konefalg) [inline whylog/client/searchers.py]: Definitely right! Let's talk about it face to face.

File: whylog/client/searchers.py (hunk 1 above)
Comment (konefalg) [inline whylog/client/searchers.py]: I agree that we should avoid using such shortcuts, but I'am not sure if `file_path` is a proper name for this variable - due to its reference to path. Maybe `opened_file` would be better?

File: whylog/tests/tests_client/test_backtrackerSearcher.py (hunk 3 above)
Comment (konefalg) [inline whylog/tests/tests_client/test_backtrackerSearcher.py]: I've tried, but yapf (command line: `yapf -ir .` in main repo directory) has reverted my changes :( So I can't aplly this proposal.

File: whylog/client/searchers.py [hunk 4]
```diff
@@ -31,6 +31,10 @@ def __init__(self, file_path):
...
         """
         a generator that returns the lines of a file in reverse order
```
Comment (kgromadzki) [inline whylog/client/searchers.py]: This docstring is out of date

File: whylog/client/searchers.py (hunk 2 above)
Comment (kgromadzki) [inline whylog/client/searchers.py]: Consider this method as classmethod

Comment (konefalg) [inline whylog/client/searchers.py]: Any reason? At the moment, this method cannot become a classmethod, due to its dependency on its field `self._file_path`.

File: whylog/tests/tests_client/test_backtrackerSearcher.py [hunk 5]
```diff
@@ -36,12 +36,20 @@ def _verify_lines(self, lines_normally, lines_reversed):
...
     ):
         backtracker = searchers.BacktrackSearcher(log_file_path)
 
         lines = self._read_last_n_lines_from_file(log_file_path, how_many_last_lines)
-        lines_reversed = list(backtracker._reverse_from_offset(*reverse_from_offset_params))
+        data_reversed = list(backtracker._reverse_from_offset(*reverse_from_offset_params))
```
Comment (ppolewicz) [inline whylog/tests/tests_client/test_backtrackerSearcher.py]: casting to list here is not necessary - it is better to let the list comprehension in the next line pull from the generator

File: whylog/tests/tests_client/test_backtrackerSearcher.py [hunk 6]
```diff
@@ -36,12 +36,20 @@ def _verify_lines(self, lines_normally, lines_reversed):
...
+        lines_reversed = [item[0] for item in data_reversed]
+
+        if final_offset_should_be_zero:
+            assert 0 == data_reversed[-1][1]
```
Comment (ppolewicz) [inline whylog/tests/tests_client/test_backtrackerSearcher.py]:
``` py
assert data_reversed[-1][1] == 0
```

By convention the constant goes to the right side of conditional expression.

File: whylog/client/searchers.py [hunk 7]
```diff
@@ -31,15 +31,21 @@ def __init__(self, file_path):
     def search(self, search_data):
         pass
 
+    @classmethod
+    def _decrease_actual_offset_properly(cls, actual_offset, decreasion_string):
```
Comment (ppolewicz) [inline whylog/client/searchers.py]: there is no such word as "decreasion"

**Pull request merged**